### PR TITLE
Compatible for windows system

### PR DESCRIPTION
**Pull request by smwgf**
Pull request for compatibility for windows system.
If system os is windows, replaced / to \\ and removed the upper case extension JPG, PNG, JPEG.

### CODE REVIEW

File: hloc/extract_features.py [hunk 1]
```diff
@@ -66,6 +66,12 @@ class ImageDataset(torch.utils.data.Dataset):
         'grayscale': False,
         'resize_max': None,
     }
+    if platform.system() =='Windows':
+        default_conf = {
+            'globs': ['*.jpg', '*.png', '*.jpeg'],
+            'grayscale': False,
+            'resize_max': None,
+        }
```
Comment (Skydes) [inline hloc/extract_features.py:74]: AFAIU glob is case-insensitive on Windows so `*.png` and `*.PNG` return identical lists. It would be easier to simply remove duplicates in `self.paths`, e.g. with
`self.paths = sorted(list(set(self.paths)))` (sorting to ensure deterministic)

File: hloc/localize_sfm.py [hunk 2]
```diff
@@ -47,15 +48,17 @@ def do_covisibility_clustering(frame_ids, all_images, points3D):
 
 def pose_from_cluster(qname, qinfo, db_ids, db_images, points3D,
                       feature_file, match_file, thresh):
-    kpq = feature_file[qname]['keypoints'].__array__()
+    if platform.system() == 'Windows':
+        kpq = feature_file[qname.replace('/', '\\')]['keypoints'].__array__()
+    else:
+        kpq = feature_file[qname]['keypoints'].__array__()
```
Comment (Skydes) [inline hloc/localize_sfm.py:54]: Do we really need to rename all query and reference paths in all these files? I would think that there are simpler options:
1. Update [`parse_retrieval`](https://github.com/cvg/Hierarchical-Localization/blob/master/hloc/utils/parsers.py#L28) to convert query and reference paths from UNIX to Windows
2. Update the [`ImageDataset`](https://github.com/cvg/Hierarchical-Localization/blob/master/hloc/extract_features.py#L110) to always return Posix paths with `path.as_posix()` instead of `str(path)`.

I would favor 2. but don't know if COLMAP parses images as Windows or Posix paths.

**Review by Skydes — CHANGES_REQUESTED** [review]
Thanks for the PR! Supporting Windows compatibility is much appreciated :) I left some comments - I think that the changes could be more minimal but don't have a windows machine to try out.

File: hloc/extract_features.py (hunk 1 above)
Comment (smwgf) [reply hloc/extract_features.py:74]: Great simple solution!! It works on my Windows system.

File: hloc/localize_sfm.py (hunk 2 above)
Comment (smwgf) [reply hloc/localize_sfm.py:54]: I will test solution 2 with "pipeline_Aachen.ipynb" script. It may take two days on my Windows system.

**Comment by smwgf** [comment]
I update the [commit](https://github.com/cvg/Hierarchical-Localization/pull/29/commits/0b6b92e810f40b624893918b7c94d49c564f4c1a).
and It works well on my Windows machine.

File: hloc/extract_features.py [hunk 3]
```diff
@@ -8,12 +8,10 @@
 import numpy as np
 from tqdm import tqdm
 import pprint
-
```
Comment (Skydes) [inline hloc/extract_features.py:11]: Great, thanks! Could you please remove all the unrelated line addition & deletion that show up in the diff? Thanks :)

**Review by Skydes — CHANGES_REQUESTED** [review]

**Comment by smwgf** [comment]
I updated the [commit ](https://github.com/cvg/Hierarchical-Localization/pull/29/commits/df41d4cecf189542ff1e7429d464508bd7d75472) to minimize changes to the [original latest commit](https://github.com/cvg/Hierarchical-Localization/commit/b9c622ae2fb48d95d8f59f887e4e1db11b095a42).

**Comment by Skydes** [comment]
Great, thank you :)
